core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -786,4 +798,8 @@ int numPendingDeletes(Index index) {
             return deleteList.size();
         }
     }
+
+    public QueryParseContext newQueryParserContext() {
```

Comment:
not sure if this belongs here, cause parsing doesn't have anything to do with indices at this point right? Maybe we should rename IndicesQueriesRegistry moving forward.